homeassistant/components/mqtt/light/schema_basic.py

```diff
@@ -441,16 +416,11 @@ def xy_received(msg):
                 return
 
             xy_color = [float(val) for val in payload.split(",")]
-            self._hs = color_util.color_xy_to_hs(*xy_color)
+            self._hs_color = color_util.color_xy_to_hs(*xy_color)
             self.async_write_ha_state()
 
         add_topic(CONF_XY_STATE_TOPIC, xy_received)
-        if (
-            self._optimistic_xy
-            and last_state
-            and last_state.attributes.get(ATTR_HS_COLOR)
-        ):
-            self._hs = last_state.attributes.get(ATTR_HS_COLOR)
+        restore_state(ATTR_HS_COLOR, self._optimistic_hs)
```

Comment:
```suggestion
        restore_state(ATTR_HS_COLOR, self._optimistic_xy)
```
Shouldn't this be `_optimistic_xy`?